assets/src/apps/authoring/components/RightMenu/RightMenu.tsx

```diff
@@ -63,6 +64,9 @@ const RightMenu: React.FC<any> = () => {
     }
     console.log('CURRENT', { currentActivity, currentLesson });
     setScreenData(transformScreenModeltoSchema(currentActivity));
+    currentActivity?.content?.partsLayout?.forEach((part: any) => {
```

Comment:
actually for these IDs we don't want duplicates from any layers either, so need to look through the whole tree.
also probably better to reduce that to a single array and call `setExisitngIds` one time instead of in a loop
```
const currentIds = currentActivityTree.reduce((acc, activity) => acc.concat(activity.content.partsLayout.map(p => p.id)), []);
```